src/umpire/strategy/DynamicPool.cpp

```diff
@@ -197,11 +196,10 @@ void* DynamicPool::allocate(std::size_t bytes)
     m_actual_bytes += alloc_bytes;
     m_curr_bytes += rounded_bytes;
 
-    const int64_t left_bytes{static_cast<int64_t>(
-        m_min_alloc_bytes - rounded_bytes)};
+    const std::size_t left_bytes{alloc_bytes - rounded_bytes};
```

Comment:
Without this change, aren't allocations greater than `m_min_alloc_bytes` problematic?  It seems that this would also cause problems.  Are we able to test this somehow?

Reply:
No. Allocations greater than `m_min_alloc_bytes` are exactly sized, so left_bytes would be > 0. In this case, left_bytes will now be 0. In any case, neither of these values would trigger the old `left_bytes > m_align_bytes` condition (which is correct).

Reply:
um..  okay?

So, this caused a problem because the (evidently small) leak would cause the containing block to become hostage and would not ever be coallesced or released.  Is that right?

Reply:
Yes - then repeating this over many many many allocations caused the large amounts we were seeing.